contracts/staking/TokenStaking.sol

```diff
@@ -0,0 +1,1228 @@
+    ///         has been authorized for at least one application. If this
+    ///         function returns true, off-chain client of the given operator is
+    ///         eligible to join the network.
+    function hasStakeDelegated(address operator)
+        external
+        view
+        override
+        returns (bool)
+    {
+        return operators[operator].authorizedApplications.length > 0;
+    }
+
+    /// @notice Returns length of application array
+    function getApplicationsLength() external view returns (uint256) {
+        return applications.length;
+    }
+
+    /// @notice Requests decrease of the authorization for the given operator on
+    ///         the given application by the provided amount.
+    ///         It may not change the authorized amount immediatelly. When
+    ///         it happens depends on the application. Can only be called by the
+    ///         given operator’s authorizer. Overwrites pending authorization
+    ///         decrease for the given operator and application.
+    /// @dev Calls `authorizationDecreaseRequested(address operator, uint256 amount)`
+    ///      on the given application. See `IApplication`.
+    function requestAuthorizationDecrease(
+        address _operator,
+        address _application,
+        uint96 _amount
+    ) public override onlyAuthorizerOf(_operator) {
+        ApplicationInfo storage application = applicationInfo[_application];
+        require(!application.disabled, "Application is disabled");
+
+        require(
+            _amount > 0,
+            "Amount to decrease authorization must greater than 0"
+        );
+
+        AppAuthorization storage authorization = operators[_operator]
+            .authorizations[_application];
+        require(
+            authorization.authorized >= _amount,
+            "Amount to decrease authorization must be less than authorized"
+        );
+
+        authorization.deauthorizing = _amount;
+        emit AuthorizationDecreaseRequested(_operator, _application, _amount);
+        IApplication(_application).authorizationDecreaseRequested(
+            _operator,
+            _amount
+        );
+    }
+
+    /// @notice Returns minimum possible stake for T, KEEP or NU in T denomination
+    function getMinStaked(address _operator, StakingProvider stakingProviders)
+        public
+        view
+        returns (uint96)
+    {
+        OperatorInfo storage operator = operators[_operator];
+        uint256 maxAuthorization = 0;
+        for (uint256 i = 0; i < operator.authorizedApplications.length; i++) {
+            address application = operator.authorizedApplications[i];
+            maxAuthorization = Math.max(
+                maxAuthorization,
+                operator.authorizations[application].authorized
+            );
+        }
+
+        if (maxAuthorization == 0) {
+            return 0;
+        }
+        if (stakingProviders != StakingProvider.T) {
```

Comment:
Can you clarify why the comparison is with `!=`? I don't understand.

Reply:
This is now explained in docs to this function, resolving.